cpp/dolfinx/io/xdmf_meshtags.h

```diff
@@ -32,11 +32,27 @@ void add_meshtags(MPI_Comm comm, const mesh::MeshTags<T>& meshtags,
   assert(meshtags.mesh());
   std::shared_ptr<const mesh::Mesh> mesh = meshtags.mesh();
   const int dim = meshtags.dim();
+  const int local_range = mesh->topology().index_map(dim)->size_local();
   const std::vector<std::int32_t>& active_entities = meshtags.indices();
+  const std::vector<std::int32_t>& values = meshtags.values();
+
+  // Remove ghost indices (note that indices are sorted)
+  int num_local_entities = 0;
```

Comment:
You can find it without the loop,
```c++
const int num_local_entities = std::lower_bound(active_entities.begin(), active_entities.end(), local_range) - active_entities.begin();
```

Just note below you are copying both vectors into another vectors, just in order to cut their tails. Maybe there is a clever way how to avoid this copy. In theory you can prepare Eigen::Map and map only the active part, and pass just the eigen map to add_topology_data. But it would need API change there.

Any better ideas @garth-wells ?

Reply:
Could just pass begin/end iterators. Agree that copying a big chuck of the vector isn't very nice.

Reply:
Any suggestions on how we could change:
```
xdmf_utils::add_data_item(attribute_node, h5_id, path_prefix + "/Values",
                            local_values, offset, {global_num_values, 1}, "",
                            use_mpi_io);
```
to use iterators? As this is a templated function, changing it to iterators is not straightforward as far as I can tell.

Reply:
@jorgensd @garth-wells I've made an attempt with Eigen::Map, see https://github.com/FEniCS/dolfinx/tree/michal/write_owned_entities

@jorgensd If you are happy with this then just extend some existing test, so it would catch the issue being addressed here.